Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerColorAnnotator.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2000-2015 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jetbrains.lang.dart.analyzer;
+
+import com.google.dart.server.generated.types.HighlightRegionType;
+import com.intellij.lang.annotation.AnnotationHolder;
+import com.intellij.lang.annotation.ExternalAnnotator;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.colors.TextAttributesKey;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiFile;
+import com.jetbrains.lang.dart.analyzer.DartAnalysisServerService.PluginHighlightRegion;
+import com.jetbrains.lang.dart.highlight.DartSyntaxHighlighterColors;
+import com.jetbrains.lang.dart.resolve.DartResolver;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class DartAnalysisServerColorAnnotator extends ExternalAnnotator<Object, Object> {
+  @Nullable
+  private static String getHighlightType(String type) {
+    if (type.equals(HighlightRegionType.ANNOTATION)) {
+      return DartSyntaxHighlighterColors.DART_METADATA;
+    }
+    if (type.equals(HighlightRegionType.BUILT_IN)) {
+      return DartSyntaxHighlighterColors.DART_KEYWORD;
+    }
+    if (type.equals(HighlightRegionType.CLASS)) {
+      return DartSyntaxHighlighterColors.DART_CLASS;
+    }
+    if (type.equals(HighlightRegionType.CONSTRUCTOR)) {
+      return DartSyntaxHighlighterColors.DART_CONSTRUCTOR_DECLARATION;
+    }
+    if (type.equals(HighlightRegionType.DYNAMIC_TYPE)) {
```

Comment:
What are arguments for having DYNAMIC_TYPE for not typed local vars and parameters instead of LOCAL_VARIABLE_DECLARATION, LOCAL_VARIABLE and PARAMETER?